modules/persistence/src/services/metadata/associated_products/index.ts

```diff
@@ -147,18 +160,45 @@ export const mergeAssociatedToCs = async (metadata) => {
   const isStagingBranch = await !getRepoBranchesEntry(project, branch);
   if (isStagingBranch) return;
 
+  const umbrellaRepoBranchesEntry = await getRepoBranchesEntry(umbrellaMetadata.project, umbrellaMetadata.branch);
+  if (!umbrellaRepoBranchesEntry)
+    throw `No repoBranches entry available for umbrella metadata with project: ${umbrellaMetadata.project}, branch: ${umbrellaMetadata.branch}`;
+
   const repoBranchesEntries = await getAllAssociatedRepoBranchesEntries(umbrellaMetadata);
   const repoBranchesMap = mapRepoBranches(repoBranchesEntries);
-  const tocsCursor = await getAssociatedProducts(umbrellaMetadata);
-  const { tocInsertions, tocOrderInsertions } = await shapeToCsCursor(tocsCursor, repoBranchesMap);
-  const mergedMetadataEntry = traverseAndMerge(umbrellaMetadata, tocInsertions, tocOrderInsertions);
+  const metadataCursor = await getAssociatedProducts(umbrellaMetadata);
```

Comment:
minor nitpick. combination of `getAssociatedProducts` and `shapeToCsCursor` can be cleaned up here a bit.

`getAssociatedProducts` can iterate over the cursor and return an object in the same format as mapRepoBranches {project: {branch: metadata}}, and `shapeToCsCursor` can read from this format and be agnostic of keys, ie. `most_recent`

Reply:
I've tinkered with this a bit and `shapeToCsCursor` can be agnostic of `most_recent` and `_id`, but the overall format of the object from `getAssociatedProducts` shouldn't match `mapRepoBranches` output, since we basically just want to match on entries that have matching keys for the `project` and `branch` values for each entry in the cursor. Making `project` and `branch` be the actual keys makes this harder - (and we'll still need to iterate over each entry from the aggregation anyways)

If we move to `{project: {branch: metadata}}`, we can pull the keys out by invoking `Object.keys` twice, but this seems redundant - or I guess we could do some fancy `...` logic, but I'm having a hard time coming up with what that would look like off the top of my head (and thus suspect that it would be non-obvious to other contributors what the codeblock is doing)

I've instead opted to destructure the bits we care about out of the `doc` and I've added a new interface to buff up the typing here a bit.

If I'm misunderstanding the suggestion, please flag so I can understand better 😄

Reply:
discussed offline. can optimize in future but not necessary ATM